src/app/core/components/language-mismatch/language-mismatch.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component } from '@angular/core';
+import { FetchState } from 'src/app/shared/helpers/state';
+import { CurrentUserHttpService } from 'src/app/shared/http-services/current-user.service';
+import { mapToFetchState } from 'src/app/shared/operators/state';
+import { LocaleService } from '../../services/localeService';
+
+@Component({
+  selector: 'alg-language-mismatch',
+  templateUrl: './language-mismatch.component.html',
+  styleUrls: [ './language-mismatch.component.scss' ],
+})
+export class LanguageMismatchComponent {
+
+  languageMismatch$ = this.localeService.languageMismatch$
+  updateUserState: FetchState<object> | undefined;
+
+  constructor(
+    private localeService: LocaleService,
+    private currentUserService: CurrentUserHttpService,
+  ) { }
+
+  onUpdateUserLanguage(language: string | undefined): void {
+    if (!language) return;
+    this.currentUserService.update({ default_language: language })
+      .pipe(mapToFetchState())
```

Comment:
`mapToFetchState` is made for fetching, not for action... probably does not fit the need here as it does not have the same state cycle as an action.

Wow wow who have written `  update(changes: object): Observable<any> `? I missed it during the review. 😣 There should not be any "any" nor "object" in the project.
Probably I could solve it separately.